api/provider/aws/retry.go

```diff
@@ -0,0 +1,22 @@
+package aws
+
+import (
+	"time"
+
+	"github.com/quintilesims/layer0/common/errors"
+)
+
+func retry(timeout, tick time.Duration, fn func() error) error {
+	after := time.After(timeout)
```

Comment:
this is a memory leak; there's no mechanism to stop the ticker's goroutine.

Reply:
Can you elaborate on this or point me to somewhere I can better understand.

Reply:
(I put my first comment in the wrong spot, this is should be on the line `case <-time.Tick(tick)`)

There's a couple things wrong. First, you aren't using `time.Tick` correctly here. You're re-creating a new `time.Ticker` instance each time we go through the `for` loop, when you really only need one.
Second, each `time.Tick` spawns its own goroutine, and you haven't provided a way to terminate those goroutines, hence the memory leak.

Reply:
Instead of trying to plug these memory leaks, I'd recommend trying a different approach to this problem. We've already established a simple pattern with the `testutils.WaitFor` helper function. I'd recommend starting there.